### PR TITLE
chore(ci): notify via slack when a publishing workflow fails

### PR DESCRIPTION
**Pull request by ArtieReus**
# Summary

To ensure we're notified of any publishing failures, integrate Slack into the workflows so that a notification is posted whenever the workflow fails. This is crucial because workflows that react to a push on the main branch run after the PR is merged. At that point, there is no active control by the developer, and if the developer isn't self-checking the action logs, any issues that arise could go unnoticed. With Slack notifications, we can proactively alert the team of any failures, ensuring that problems are addressed immediately.

# Changes Made

- create a reuse workflow for the slack notification
- notify on failure when building supernova
- notify on failure when building greenhouse
- notify on success on npm release using the reuse workflow

# Related Issues

- #409

# Screenshots (if applicable)

# Testing Instructions

1. `npm i`
2. `npm run TASK`

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

**Comment by ArtieReus** [comment]
fixes #409

**Review by hodanoori — APPROVED** [review]

File: .github/workflows/build-push-greenhouse-image.yaml [hunk 1]
```diff
@@ -154,3 +145,13 @@ jobs:
         # This step uses the identity token to provision an ephemeral certificate
         # against the sigstore community Fulcio instance.
         run: echo "${TAGS}" | xargs -I {} cosign sign --yes {}@${DIGEST}
+
+  notify-on-failure:
+    if: failure()
+    needs: [build-and-push-greenhouse-image]
+    uses: cloudoperators/juno/.github/workflows/shared-slack-notification.yaml@main
+    with:
+      title: "🚨 JUNO Supernova Image Failed 🚨"
```
Comment (hodanoori) [inline .github/workflows/build-push-greenhouse-image.yaml:154]: Same here, Greenhouse instead of Supernova?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -42,45 +45,23 @@ jobs:
...
+      rich_body: true
+    secrets:
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+  notify-on-failure:
+    if: failure()
+    needs: [release]
+    uses: cloudoperators/juno/.github/workflows/shared-slack-notification.yaml@artie-shared-slack
+    with:
+      title: "🚨 JUNO Supernova Image Failed 🚨"
```
Comment (hodanoori) [inline .github/workflows/release.yaml:64]: Why only Supernova? Shouldn't be JUNO package?

**Review by barsukov — CHANGES_REQUESTED** [review]
Lets us fix errors and merge it

File: .github/workflows/build-push-greenhouse-image.yaml (hunk 1 above)
Comment (ArtieReus) [reply .github/workflows/build-push-greenhouse-image.yaml:154]: Thx! @hodanoori

File: .github/workflows/release.yaml (hunk 2 above)
Comment (ArtieReus) [reply .github/workflows/release.yaml:64]: Thx! @hodanoori

File: .github/workflows/build-push-greenhouse-image.yaml [hunk 3]
```diff
@@ -154,3 +145,13 @@ jobs:
...
+      body: "An error occurred while building the Supernova image for the branch ${{ github.head_ref || github.ref_name }}. Please check the logs for more information. <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|Check the logs>"
```
[the remark this answers is not in the document]
Comment (ArtieReus) [reply .github/workflows/build-push-greenhouse-image.yaml]: Thx! @hodanoori

**Review by barsukov — APPROVED** [review]
Much better! thank you 💯

**Review by andypf — APPROVED** [review]

**Pull request merged**